src/callTraceStorage.cpp

```diff
@@ -78,8 +78,11 @@ class LongHashTable {
     }
 };
 
-
+#if defined(__x86_64__) || defined(__aarch64__) || (defined(__riscv) && (__riscv_xlen == 64))
```

Comment:
There is a built-in `_LP64` macro that better fits here.

Also, to avoid ifdefs in shared code and make it overall cleaner, I'd suggest to define `LP64_ONLY()` macro in arch.h, similar to how HotSpot does it. E.g.
```
CallTrace CallTraceStorage::_overflow_trace = {1, {BCI_ERROR, LP64_ONLY(0,) (jmethodID)"storage_overflow"}};
...
LP64_ONLY(frames[i].padding = 0;)
```
etc.

Reply:
pushed, also changed malloc to calloc.